### PR TITLE
#8 Create component: button, input, heading

### PR DESCRIPTION
**Pull request by DuongPham-dev**
- #8

### CODE REVIEW

File: practices/practice-one/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  tag?: "button" | "a";
+  path?: string;
+  variant?: "primary" | "secondary" | "default";
+  size?: "sm" | "md" | "lg";
+  borderRadius?: "5px" | "10px" | "15px" | "25px" | "50%";
+  children?: ReactNode;
+  onClick?: ClickEvent;
+}
+
+const buttonStize = {
```
Comment (huypham-agilityio) [inline practices/practice-one/src/components/Button/Button.tsx]: typing mistake

File: practices/practice-one/src/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    tag = "button",
+    size = "sm",
+    variant = "default",
+    path = "",
+    children,
+    ...rest
+  } = props;
+  const TagName = tag === "a" ? NavLink : tag;
+  const className = `btn btn-${variant} btn-${buttonStize[size]}`;
+  return (
```
Comment (huypham-agilityio) [inline practices/practice-one/src/components/Button/Button.tsx:39]: code style. add empty line for similar places

File: practices/practice-one/src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { ReactNode, MouseEvent } from "react";
+import { NavLink } from "react-router-dom";
+import "./button.css";
+type ClickEvent = (event: MouseEvent) => void;
+
+interface ButtonProps {
```
Comment (huypham-agilityio) [inline practices/practice-one/src/components/Button/Button.tsx:7]: Disabled state of button?

File: practices/practice-one/src/components/Input/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { ChangeEvent } from "react";
+import "./input.css";
+
+type CustomStyle = {
+  [key: string]: string;
+};
+
+interface InputProps {
```
Comment (huypham-agilityio) [inline practices/practice-one/src/components/Input/Input.tsx]: it's better to extends props from InputHTMLAttributes

Comment (DuongPham-dev) [reply practices/practice-one/src/components/Input/Input.tsx]: I added it

File: practices/practice-one/src/components/Button/Button.tsx (hunk 1 above)
Comment (DuongPham-dev) [reply practices/practice-one/src/components/Button/Button.tsx]: I've updated

File: practices/practice-one/src/components/Button/Button.tsx (hunk 2 above)
Comment (DuongPham-dev) [reply practices/practice-one/src/components/Button/Button.tsx:39]: I've updated

File: practices/practice-one/src/components/Button/Button.tsx (hunk 3 above)
Comment (DuongPham-dev) [reply practices/practice-one/src/components/Button/Button.tsx:7]: I don't understand this part, can you explain more?

Comment (huypham-agilityio) [reply practices/practice-one/src/components/Button/Button.tsx:7]: should have **disabled** prop

File: practices/practice-one/src/components/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  type?: "button" | "submit";
+  title?: string;
+  tag?: "button" | "a";
+  path?: string;
+  variant?: "primary" | "secondary" | "default";
+  size?: "sm" | "md" | "lg";
+  borderRadius?: "5px" | "10px" | "15px" | "25px" | "50%";
+  disabled: boolean;
```
Comment (huypham-agilityio) [inline practices/practice-one/src/components/Button/Button.tsx]: It's better to optional

**Review by huypham-agilityio — APPROVED** [review]

**Pull request merged**